hathor/indexes/rocksdb_tokens_index.py

```diff
@@ -0,0 +1,485 @@
+    def _create_genesis_info(self) -> None:
+        self._create_token_info(
+            settings.HATHOR_TOKEN_UID,
+            settings.HATHOR_TOKEN_NAME,
+            settings.HATHOR_TOKEN_SYMBOL,
+            settings.GENESIS_TOKENS,
+        )
+
+    def _add_to_total(self, token_uid: bytes, amount: int) -> None:
+        key_info = self._to_key_info(token_uid)
+        old_value_info = self._db.get((self._cf, key_info))
+        if token_uid == settings.HATHOR_TOKEN_UID and old_value_info is None:
+            self._create_genesis_info()
+            old_value_info = self._db.get((self._cf, key_info))
+        assert old_value_info is not None
+        dict_info = self._from_value_info(old_value_info)
+        dict_info['total'] += amount
+        new_value_info = self._to_value_info(dict_info)
+        self._db.put((self._cf, key_info), new_value_info)
+
+    def _subtract_from_total(self, token_uid: bytes, amount: int) -> None:
+        key_info = self._to_key_info(token_uid)
+        old_value_info = self._db.get((self._cf, key_info))
+        if token_uid == settings.HATHOR_TOKEN_UID and old_value_info is None:
+            self._create_genesis_info()
+            old_value_info = self._db.get((self._cf, key_info))
+        assert old_value_info is not None
+        dict_info = self._from_value_info(old_value_info)
+        dict_info['total'] -= amount
+        new_value_info = self._to_value_info(dict_info)
+        self._db.put((self._cf, key_info), new_value_info)
+
+    def _add_authority_utxo(self, tx: BaseTransaction, index: int) -> None:
+        """ Add tx to mint/melt indexes and total amount
+        """
+        assert tx.hash is not None
+
+        tx_output = tx.outputs[index]
+        token_uid = tx.get_token_uid(tx_output.get_token_index())
+
+        if tx_output.is_token_authority():
+            if tx_output.can_mint_token():
+                # add to mint index
+                self._add_mint_utxo(token_uid, tx.hash, index)
+            if tx_output.can_melt_token():
+                # add to melt index
+                self._add_melt_utxo(token_uid, tx.hash, index)
+        else:
+            self._add_to_total(token_uid, tx_output.value)
+
+    def _remove_authority_utxo(self, tx: BaseTransaction, index: int) -> None:
+        """ Remove tx from mint/melt indexes and total amount
+        """
+        assert tx.hash is not None
+
+        tx_output = tx.outputs[index]
+        token_uid = tx.get_token_uid(tx_output.get_token_index())
+
+        if tx_output.is_token_authority():
+            if tx_output.can_mint_token():
+                # remove from mint index
+                self._remove_mint_utxo(token_uid, tx.hash, index)
+            if tx_output.can_melt_token():
+                # remove from melt index
+                self._remove_melt_utxo(token_uid, tx.hash, index)
+        else:
+            self._subtract_from_total(token_uid, tx_output.value)
+
+    def add_tx(self, tx: BaseTransaction) -> None:
```

Comment:
From an initial analysis, both `add_tx` and `del_tx` could be refactored to the base class, with specialized methods for handling the underlying storage in each subclass.

I'm not sure if it's worth doing it now or in a future PR.

Reply:
I guess they could, but then the base class would have to abstract some more "internal" methods. I'm not entirely sure if it's worth it in this case.